### PR TITLE
feature/ka-278-most-common-errors

### PR DESCRIPTION
**Pull request by drcgjung**
- introduce warning channel into the agent plane (header in matchmaking, body part in transfer)
- deal with warnings as alerts in skill framework and asset view as well as sparql editor.

### CODE REVIEW

File: ux/skill_framework/src/remote_connector.ts [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        };
+        return errorResult;
+      }
+
+      //result: BindingSet
+      const result = (await response.json()) as BindingSet;
+
+      if (response.headers.has('cx_warnings')) {
+        const warnings = JSON.parse(
+          response.headers.get('cx_warnings')!
```
Comment (JaneMcBrain) [inline ux/skill_framework/src/remote_connector.ts:115]: you can write `response.headers.get('cx_warnings')` || ''`

File: ux/skill_framework/src/remote_connector.ts [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
```
Comment (JaneMcBrain) [inline ux/skill_framework/src/remote_connector.ts:217]: same here: `response.headers.get('cx_warnings') || '`

File: ux/skill_framework/src/data.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+// See copyright notice in the top folder
+// See authors file in the top folder
+// See license file in the top folder
+//
+
+/**
+ * some typescript pseudo-bindings
+ * for generic JSON objects
+ */
+export type JSONElement = JSONObject | JSONArray;
```
Comment (JaneMcBrain) [inline ux/skill_framework/src/data.ts:12]: its more common to name this file to types.ts  - data might be misleading

**Review by JaneMcBrain — APPROVED** [review]

File: ux/skill_framework/src/data.ts (hunk 3 above)
Comment (drcgjung) [reply ux/skill_framework/src/data.ts:12]: hmm, but we also may have interface_types ... lets look at this next week.

File: ux/skill_framework/src/remote_connector.ts (hunk 1 above)
Comment (drcgjung) [reply ux/skill_framework/src/remote_connector.ts:115]: Im a fan of explicit code ;-)

**Pull request merged**